airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/destination_async/FlushWorkersTest.java

```diff
@@ -25,15 +25,22 @@ public class FlushWorkersTest {
 
   @Test
   void testErrorHandling() throws Exception {
+    final AtomicBoolean hasThrownError = new AtomicBoolean(false);
     final var desc = new StreamDescriptor().withName("test");
     final var dequeue = mock(BufferDequeue.class);
     when(dequeue.getBufferedStreams()).thenReturn(Set.of(desc));
-    when(dequeue.take(desc, 1000)).thenReturn(new MemoryAwareMessageBatch(List.of(), 0, null, null));
-    when(dequeue.getQueueSizeInRecords(desc)).thenReturn(Optional.of(1L));
-    final var collector = mock(Consumer.class);
+    when(dequeue.take(desc, 1000)).thenReturn(new MemoryAwareMessageBatch(List.of(), 10, null, null));
+    when(dequeue.getQueueSizeBytes(desc)).thenReturn(Optional.of(10L));
+    when(dequeue.getQueueSizeInRecords(desc)).thenAnswer(ignored -> {
+      if (hasThrownError.get()) {
+        return Optional.of(0L);
+      } else {
+        return Optional.of(1L);
+      }
+    });
```

Comment:
for understanding: is this trying to force an exception in the supervisor thread?

Reply:
nope. i changed the behavior in the worker so that it's not allowed to close until all the queues are empty. so now in this test, i can't let the queue go to zero until after we throw, otherwise the workers can close before the exception gets thrown.